cmd/ebrelayer/txs/relayToEthereum.go

```diff
@@ -67,52 +49,31 @@ func RelayProphecyClaimToEthereum(provider string, contractAddress common.Addres
 		"CosmosSender", claim.CosmosSender,
 		"CosmosSenderSequence", claim.CosmosSenderSequence)
 
-	tx, err := cosmosBridgeInstance.NewProphecyClaim(auth, uint8(claim.ClaimType),
+	tx, _ := cosmosBridgeInstance.NewProphecyClaim(auth, uint8(claim.ClaimType),
 		claim.CosmosSender, claim.CosmosSenderSequence, claim.EthereumReceiver, claim.Symbol, claim.Amount.BigInt())
-
-	if err != nil && err.Error() == "not found" {
-		// if there is a not found error while broadcasting to infura, retry until there isn't an error
-		i := 0
-		for i < 30 {
-			sugaredLogger.Errorw(
-				"could not find tx after broadcasting",
-				"retry", i,
-				"err", err.Error(),
-			)
-
-			tx, err = cosmosBridgeInstance.NewProphecyClaim(auth, uint8(claim.ClaimType),
-				claim.CosmosSender, claim.CosmosSenderSequence, claim.EthereumReceiver, claim.Symbol, claim.Amount.BigInt())
-			if err == nil {
-				break
-			}
-
-			// sleep for 1 second until retrying
-			time.Sleep(transactionInterval)
-			i++
-		}
-
-		if i == 30 {
-			sugaredLogger.Errorw("failed to send ProphecyClaim to CosmosBridge.",
-				errorMessageKey, err.Error())
-			return err
-		}
-	}
+	// sleep 30 seconds to wait for tx to go through.
+	time.Sleep(transactionInterval * 30)
 
 	sugaredLogger.Infow("get NewProphecyClaim tx hash:", "ProphecyClaimHash", tx.Hash().Hex())
 
+	txHash := tx.Hash()
+
 	// Get the transaction receipt
-	receipt, err := client.TransactionReceipt(context.Background(), tx.Hash())
-	if err != nil {
+	receipt, err := client.TransactionReceipt(context.Background(), txHash)
+	// if there is an error getting the tx, or if the tx fails, retry 30 times
+	if err != nil || receipt.Status == 0 {
 		i := 0
 		for i < 30 {
 			sugaredLogger.Errorw(
 				"no tx receipt after broadcasting",
 				"retry", i,
 				"getTxReceiptErr", err.Error(),
 			)
+			_, _ = cosmosBridgeInstance.NewProphecyClaim(auth, uint8(claim.ClaimType),
+				claim.CosmosSender, claim.CosmosSenderSequence, claim.EthereumReceiver, claim.Symbol, claim.Amount.BigInt())
 
-			receipt, err := client.TransactionReceipt(context.Background(), tx.Hash())
-			if err == nil {
+			receipt, err := client.TransactionReceipt(context.Background(), txHash)
+			if err == nil || receipt.Status == 1 {
```

Comment:
I don't think we can do this.  If err is set, there's no promise that receipt is set to anything reasonable, up to and including nil